views/settings/index.js

```diff
@@ -102,36 +140,106 @@ export default class SettingsView extends React.Component {
-        isDisabled={disabled}
-        onPress={loggedIn ? this.logOut : this.logIn}
+        isDisabled={disabledGoogle}
+        onPress={loggedInGoogle ? this.logOutGoogle : this.logInGoogle}
+      >
+        <Text style={[styles.loginButtonText, loginTextStyleGoogle]}>
+          {loadingGoogle
+            ? 'Logging in to Google…'
+            : loggedInGoogle
+              ? 'Sign Out of Google'
+              : 'Sign In with Google'}
+        </Text>
+      </CustomCell>
+    )
+
+    let usernameCell = (
+      <CustomCell contentContainerStyle={styles.loginCell}>
+        <Text onPress={this.focusUsername} style={styles.label}>Username</Text>
+        <TextInput
+          ref={ref => this._usernameInput = ref}
+          cellStyle='Basic'
+          autoCorrect={false}
+          autoCapitalize='none'
+          style={styles.customTextInput}
+          placeholderTextColor='#C7C7CC'
+          disabled={disabledStOlaf}
+          placeholder='username'
+          value={username}
+          returnKeyType='next'
+          onChangeText={text => this.setState({username: text})}
+          onSubmitEditing={this.focusPassword}
+        />
+      </CustomCell>
+    )
+
+    let passwordCell = (
+      <CustomCell contentContainerStyle={styles.loginCell}>
+        <Text onPress={this.focusPassword} style={styles.label}>Password</Text>
+        <TextInput
+          ref={ref => this._passwordInput = ref}
+          cellStyle='Basic'
+          autoCorrect={false}
+          autoCapitalize='none'
+          style={styles.customTextInput}
+          placeholderTextColor='#C7C7CC'
+          disabled={disabledStOlaf}
+          secureTextEntry={true}
+          placeholder='password'
+          value={password}
+          returnKeyType='done'
+          onChangeText={text => this.setState({password: text})}
+          onSubmitEditing={loggedInStOlaf ? () => {} : this.logInStOlaf}
+        />
+      </CustomCell>
+    )
+
+    let loginButton = (
+      <CustomCell
+        contentContainerStyle={styles.actionButton}
+        isDisabled={disabledStOlaf}
+        onPress={loggedInStOlaf ? this.logOutStOlaf : this.logInStOlaf}
       >
-        <Text style={[styles.loginButtonText, loginTextStyle]}>
-          {loading
-            ? 'Logging in…'
-            : loggedIn
-              ? 'Sign Out'
-              : 'Sign In'}
+        <Text style={[styles.loginButtonText, loginTextStyleStOlaf]}>
+          {loadingStOlaf
+            ? 'Logging in St. Olaf…'
+            : loggedInStOlaf
+              ? 'Sign Out of St. Olaf'
+              : 'Sign In to St. Olaf'}
         </Text>
       </CustomCell>
     )
 
     let accountSection = (
       <View>
-        <Section>
+        <Section header='LOGIN' footer='Note: This application requires logging in to Google and inputting your St. Olaf username and password in order use all features.'>
+          {usernameCell}
+          {passwordCell}
+          {loginButton}
           {actionCell}
```

Comment:
I don't like having both of the login buttons right next to each other. Can we pull the "Sign In with Google" button into its own `<Section/>`?